### ISSUE TITLE
Remove the spaces between each number for the telephone form

### ISSUE THREAD

**Issue by gobyrne** [GCTC-NTGC/gc-digital-talent#7769]
## Details
Some voice to text users are unable enter a phone number (in our phone number input) because there are spaces between the numbers that they dictate. To allow for more flexibility we should remove all white space from the number before saving it to the database.

## Acceptance criteria

- [ ] Phone numbers with spaces should be saved gracefully. For example: "1 2 1 1 2 1 1 1 2 1 "

**Comment by petertgiles** [comment]
I'm a little confused by this one.  Currently we have no validation on our phone numbers and they can be saved with spaces:
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/69d449b9-1b5f-4570-8b5c-5d82d7e433d7)

**Comment by gobyrne** [comment]
Thanks @petertgiles , you're right that it wasn't what I described here. We tried again with #7769

Closing this one

**Comment by gobyrne** [comment]
Hm... I guess we don't understand what the problem was.

@RM-1978 @esizer , any ideas?

**Comment by gobyrne** [comment]
You can get more details in the issue history

**Comment by esizer** [comment]
> Hm... I guess we don't understand what the problem was.

Yeah, now that I see @petertgiles comments I don't really have appetite to pursue this one. I feel like removing whitespace is reducing flexibility and kind of mangles specific formats that people may prefer. 

I'd say, unless we want to enforce a specific format (I personally don't want to) we should leave this field fairly open. Phone numbers are hard to deal with because of many different reasons, format being one of them.
